### ISSUE TITLE
WHAM ConvTasnet recipe 4.5dB worse than reported numbers

### ISSUE THREAD

**Issue by rohitpaturi** [mpariente/asteroid#188]
I have tried the Wham ConvTasnet recipe and the SI-SDR comes out to be 11.9dB whereas the reported numbers are 16.2dB, which is a huge gap. So, I am wondering what is wrong with my setup and config.
CONFIG:
(I have tried  different configs. Pasting the best config here)
data:
  mode: min
  nondefault_nsrc: null
  sample_rate: 8000
  task: sep_clean
  train_dir: data/wav8k/min/tr
  valid_dir: data/wav8k/min/cv
filterbank:
  kernel_size: 16
  n_filters: 512
  stride: 8
main_args:
  exp_dir: exp/train_convtasnet/
  help: null
masknet:
  bn_chan: 128
  hid_chan: 512
  mask_act: relu
  n_blocks: 8
  n_repeats: 3
  n_src: 2
  skip_chan: 128
optim:
  lr: 0.001
  optimizer: adam
  weight_decay: 0.0
positional arguments: {}
training:
  batch_size: 24
  early_stop: true
  epochs: 200
  half_lr: true
  num_workers: 8

**Comment by mpariente** [comment]
Can you listen to the clean speech of wsj0? 
It might seem like you generated them from wv1 instead of wv0. In such case, there is small background noise in the clean data.

**Comment by rohitpaturi** [comment]
Oh okay, thanks for the information! Yes, I did generate the waveforms from wv1. Do you mean wv2 instead of wv0. As far as I know, WSJ0 only comes with only wv[1-2] extensions, right?

**Comment by mpariente** [comment]
Oh yes sorry, should be generated with wv1, not wv2, so you did right. 
Did you look at the training log, how does it look like?

**Comment by mpariente** [comment]
Can you also run evaluation from a pretrained model to validate that you have the good result?

**Comment by rohitpaturi** [comment]
The training log looks okay, but with the loss saturating around the 150th epoch.
I've tried the pre-trained model published here: https://zenodo.org/record/3862942#.XydLZfhKhQI and here are the results I get:
Overall metrics :
{'sar': 10.44097769122019,
 'sar_imp': -138.711218817442,
 'sdr': 8.450281738658656,
 'sdr_imp': 8.146553121444242,
 'si_sdr': 7.492973677180067,
 'si_sdr_imp': 7.4944194836328,
 'sir': 15.377501845990885,
 'sir_imp': 15.073773228776457,
 'stoi': 0.794393625040599,
 'stoi_imp': 0.1356283198136183}

I am certain that there is some mismatch in the data preparation but, on listening to the mixed samples, they do sound clean mixed. I am not sure exactly what the issue is. I am trying to generate the data from scratch (I had pre-generated wham data so far) but previously when I tried preparing the data, it fails in the "Stage 1: Generating 8k and 16k WHAM dataset". I will look into this and get back(in another issue) if needed.

**Comment by mpariente** [comment]
Can you listen to the clean files directly? 
Please try to generate from scratch yes, the evaluation scores don't look right.

**Comment by rohitpaturi** [comment]
Thanks! I could replicate the results preparing the data from scratch. The reason for the degradation was due to data prepared from .wv2 instead of .wv1. Closing this issue.

**Comment by mpariente** [comment]
Great to hear it ! 
Refs #166 #164
